.github/workflows/udapte-deps.yml

```diff
@@ -0,0 +1,56 @@
+name: 'Update Nix Flake Inputs'
+on:
+  push:
+    branches:
+      - '_update-deps_runtimeverification_haskell-backend'
+      - '_update-deps_runtimeverification_llvm-backend'
+  workflow_dispatch:
+# Stop in progress workflows on the same branch and same workflow to use latest committed code
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  nix-flake-submodule-sync:
+    name: 'Nix flake submodule sync'
+    runs-on: ubuntu-latest
+    steps:
+      - id: config
+        run: |
+          ref=${{ github.ref }}
+          if [ "${{ github.event_name }}" == 'pull_request' ]; then
+            ref="${{ github.event.pull_request.head.sha }}"
+          fi
+          ref="${ref#refs/heads/}"
+          echo "::set-output name=ref::$ref"
```

Comment:
This is also basically fully unnecessary now that I read it. I'll make a separate PR cleaning up some of this stuff and the maven steps probably.

Reply:
Same, I'll make these changes in my local branch then open a PR with updates.